02-intermediate-ruby/object-inheritance.md

```diff
@@ -3,77 +3,131 @@
+For example, we could have a `LibraryItem` class that describes the general behaviors of items in a library. This is a _base class_ or _superclass_. We could create `Book` and `Video` classes that model more specific behaviors. A `Book` is very different than a `Video`, but they are both items in the library. It would make sense then, to declare them _subclasses_ of `LibraryItem`.
 
-A way to describe this is by using the _is-a_ phrase: `Penguin` _is-a_ `Bird`. `Hawk` _is-a_ `Bird`. In Ruby, we use the `<` symbol to denote that the class on the left of the `<` symbol is inheriting behaviors from the class on the right side of the `<` symbol:
+A way to describe this is by using the _is-a_ phrase: a `Book` _is-a_ `LibraryItem`. `Video` _is-a_ `LibraryItem`. In Ruby, we use the `<` symbol to denote that the class on the left of the `<` symbol is inheriting behaviors from the class on the right side of the `<` symbol:
 
 ```ruby
-class Bird
-  def talk
-    "I'm a bird. Kah."
+class LibraryItem
+  def barcode
+    return "12345"
   end
 end
 
-# Penguin is-a Bird
-# Penguin inherits from Bird
-class Penguin < Bird
+# Book is-a LibraryItem
+# Book inherits from LibraryItem
+class Book < LibraryItem
 end
 ```
 
+The code above creates an is-a relationship between `LibraryItem` and `Book`.  The relationship diagramed below gives any instance of `Book` a `barcode` method.
+
+![Inheritance Diagram](images/basicInheritance.png)
+
 ## Inheriting Methods
-So what do we get when we inherit? We get __methods__. The  `Bird` class has an instance method called `talk`. Since all birds talk, our _subclasses_ will _inherit_ this method. So now our `Penguin` instances will have the talk method.
+So what do we get when we inherit? We get __methods__. The  `Book` class has an instance method called `barcode`. Since all items in the library have a barcode, our _subclasses_ will _inherit_ this method. So now our `Book` instances will have the `barcode` method.
 
 ```ruby
-generic_bird = Bird.new
-generic_bird.talk #=> "I'm a bird. Kah."
+generic_item = LibraryItem.new
+generic_item.barcode #=> "12345"
 
-tux = Penguin.new
-tux.talk # => "I'm a bird. Kah."
+poodr = Book.new
+poodr.barcode #=> "12345"
 ```
 
-This is known as _implicit inheritance_ because when we put functionality in the _base class_ (`Bird`), then all _subclasses_ (`Penguin`) will automatically get those behaviors.
+This is known as _implicit inheritance_ because when we put functionality in the _base class_ (`LibraryItem`), then all _subclasses_ (`Book`) will automatically get those behaviors.
 
 ## Overriding Methods
 What happens when our classes have different functionality for the same idea? We can _override_ methods from _base classes_ in their _subclasses_.
```

Comment:
Could we add some more explanation about why we'd want the barcode to be differentiated from one object to another?